packages/flutter_tools/lib/src/commands/generate_localizations.dart

```diff
@@ -240,69 +232,22 @@ class GenerateLocalizationsCommand extends FlutterCommand {
-      format = format || options.format;
     } else {
-      final String inputPathString = stringArg('arb-dir')!; // Has default value, cannot be null.
-      final String? outputPathString = stringArg('output-dir');
-      final String outputFileString = stringArg('output-localization-file')!; // Has default value, cannot be null.
-      final String templateArbFileName = stringArg('template-arb-file')!; // Has default value, cannot be null.
-      final String? untranslatedMessagesFilePath = stringArg('untranslated-messages-file');
-      final String classNameString = stringArg('output-class')!; // Has default value, cannot be null.
-      final List<String> preferredSupportedLocales = stringsArg('preferred-supported-locales');
-      final String? headerString = stringArg('header');
-      final String? headerFile = stringArg('header-file');
-      final bool useDeferredLoading = boolArg('use-deferred-loading');
-      final String? inputsAndOutputsListPath = stringArg('gen-inputs-and-outputs-list');
-      final bool useSyntheticPackage = boolArg('synthetic-package');
-      final String? projectPathString = stringArg('project-dir');
-      final bool areResourceAttributesRequired = boolArg('required-resource-attributes');
-      final bool usesNullableGetter = boolArg('nullable-getter');
-      final bool useEscaping = boolArg('use-escaping');
-      final bool suppressWarnings = boolArg('suppress-warnings');
-
-      precacheLanguageAndRegionTags();
-
-      try {
-        final LocalizationsGenerator generator = LocalizationsGenerator(
-          fileSystem: _fileSystem,
-          inputPathString: inputPathString,
-          outputPathString: outputPathString,
-          templateArbFileName: templateArbFileName,
-          outputFileString: outputFileString,
-          classNameString: classNameString,
-          preferredSupportedLocales: preferredSupportedLocales,
-          headerString: headerString,
-          headerFile: headerFile,
-          useDeferredLoading: useDeferredLoading,
-          inputsAndOutputsListPath: inputsAndOutputsListPath,
-          useSyntheticPackage: useSyntheticPackage,
-          projectPathString: projectPathString,
-          areResourceAttributesRequired: areResourceAttributesRequired,
-          untranslatedMessagesFile: untranslatedMessagesFilePath,
-          usesNullableGetter: usesNullableGetter,
-          useEscaping: useEscaping,
-          logger: _logger,
-          suppressWarnings: suppressWarnings,
-        )
-          ..loadResources()
-          ..writeOutputFiles();
-        outputFileList = generator.outputFileList;
-        untranslatedMessagesFile = generator.untranslatedMessagesFile;
-      } on L10nException catch (e) {
-        throwToolExit(e.message);
-      }
+      options = parseLocalizationsOptionsFromCommand(command: this);
     }
 
+    // Run the localizations generator.
+    final LocalizationsGenerator generator = generateLocalizations(
```

Comment:
Why are we no longer catching `L10nException`? Does this get caught somewhere else?

Reply:
Yep, this already happens in `generateLocalizations`.